process/sync/storageBootstrap/baseStorageBootstrapper.go

```diff
@@ -0,0 +1,310 @@
+		highestNonce = strHdrI.HeaderInfo.Nonce
+
+		lastRound = strHdrI.LastRound
+		if lastRound == 0 {
+			break
+		}
+	}
+
+	return bootInfos, nil
+}
+
+func (st *storageBootstrapper) applyBootInfos(bootInfos []bootstrapStorage.BootstrapData) error {
+	var err error
+
+	defer func() {
+		if err != nil {
+			st.blkExecutor.RestoreLastNotarizedHrdsToGenesis()
+			st.forkDetector.RestoreFinalCheckPointToGenesis()
+		}
+	}()
+
+	for i := len(bootInfos) - 1; i >= 0; i-- {
+		log.Debug("apply block",
+			"nonce", bootInfos[i].HeaderInfo.Nonce,
+			"shardId", bootInfos[i].HeaderInfo.ShardId)
+
+		lastNotarized := make(map[uint32]*sync.HdrInfo)
+		for _, lastNotarizedHeader := range bootInfos[i].LastNotarizedHeaders {
+			log.Debug("added notarized header",
+				"nonce", lastNotarizedHeader.Nonce,
+				"shardId", lastNotarizedHeader.ShardId)
+
+			lastNotarized[lastNotarizedHeader.ShardId] = &sync.HdrInfo{
+				Nonce: lastNotarizedHeader.Nonce,
+				Hash:  lastNotarizedHeader.Hash,
+			}
+		}
+
+		err = st.bootstrapper.applyNotarizedBlocks(lastNotarized)
+		if err != nil {
+			log.Debug("cannot apply notarized block", "error", err.Error())
+
+			return err
+		}
+
+		lastFinalHashes := make([][]byte, 0)
+		for _, lastFinal := range bootInfos[i].LastFinals {
+			lastFinalHashes = append(lastFinalHashes, lastFinal.Hash)
+		}
+
+		err = st.addHeaderToForkDetector(bootInfos[i].HeaderInfo.Hash, lastFinalHashes)
+		if err != nil {
+			log.Debug("cannot apply final block", "error", err.Error())
+			return err
+		}
+	}
+
+	return nil
+}
+
+func (st *storageBootstrapper) cleanupStorage(headerInfo bootstrapStorage.BootstrapHeaderInfo) {
+	log.Debug("cleanup storage")
+
+	nonceToByteSlice := st.uint64Converter.ToByteSlice(headerInfo.Nonce)
+	err := st.headerNonceHashStore.Remove(nonceToByteSlice)
+	if err != nil {
+		log.Debug("block was not removed from storage",
+			"shradId", headerInfo.ShardId,
+			"nonce", headerInfo.Nonce,
+			"hash", headerInfo.Hash,
+			"error", err.Error())
+		return
+	}
+
+	log.Debug("block was removed from storage",
+		"shradId", headerInfo.ShardId,
+		"nonce", headerInfo.Nonce,
+		"hash", headerInfo.Hash)
+}
+
+func (st *storageBootstrapper) displayProcessedMiniBlocks(processedMiniBlocks map[string]map[string]struct{}) {
+	log.Debug("processed mini blocks applied")
+
+	for metaBlockHash, miniBlocksHashes := range processedMiniBlocks {
+		log.Debug("processed",
+			"meta hash", []byte(metaBlockHash))
+		for miniBlockHash := range miniBlocksHashes {
+			log.Debug("processed",
+				"mini block hash", []byte(miniBlockHash))
+		}
+	}
+}
+
+func (st *storageBootstrapper) getShardHeaderFromStorage(headerHash []byte) (data.HeaderHandler, error) {
+	header, err := process.GetShardHeaderFromStorage(headerHash, st.marshalizer, st.store)
+
+	return header, err
+}
+
+func (st *storageBootstrapper) getMetaHeaderFromStorage(hash []byte) (data.HeaderHandler, error) {
+	header, err := process.GetMetaHeaderFromStorage(hash, st.marshalizer, st.store)
```

Comment:
why do we need these internal wrappers, the same information can be accessed directly using the wrapped over functionality, without any change, I would remove  getMetaHeaderFromStorage and getShardHeaderFromStorage

Reply:
good catch. done.